src/main/java/org/jenkinsci/plugins/pubsub/JobChannelMessage.java

```diff
@@ -80,20 +80,19 @@ public String getJobName() {
      * Jenkins {@link Item}.
      */
     public synchronized @CheckForNull Item getJobChannelItem() {
-        if (jobLookupComplete || jobChannelItem != null) {
-            return jobChannelItem;
+        if (jobLookupComplete || jobChannelItemFullName != null) {
+            return Jenkins.get().getItemByFullName(jobChannelItemFullName);
         }
         
         try {
             String jobName = get(EventProps.Job.job_name);
             if (jobName != null) {
-                Jenkins jenkins = Jenkins.get();
-                jobChannelItem = jenkins.getItemByFullName(jobName);
```

Comment:
Already applied to the previous code so probably OK, but beware that this call may

* return null if there is no such item
* return an item of that name
* return null if there _is_ an item of that name but the current authn has no `READ` nor `DISCOVER` permission on it
* throw an exception if there is an item of that name and the current authn has `DISCOVER` but _not_ `READ` on it